### PR TITLE
refactor: replaces vm_exitcode.id with go code

### PR DESCRIPTION
**Pull request by frrist**

### CODE REVIEW

File: src/actors/runtime/exitcode/vm_exitcodes.go [hunk 1]
```diff
@@ -65,27 +65,41 @@ const (
 	InsufficientPledgeCollateral
 )
 
+type ExitCode struct {
+	Success          struct{}
+	SystemError      SystemErrorCode
+	UserDefinedError UserDefinedErrorCode
+}
```
Comment (anorth) [inline src/actors/runtime/exitcode/vm_exitcodes.go]: Thanks for this faithful port, but it's too complex. Can you replace all of this with a single `type ErrorCode int64` and some helper methods? I think we can remove the concept of UserDefinedErrorCode.

File: src/actors/runtime/exitcode/vm_exitcodes.go [hunk 2]
```diff
@@ -65,56 +64,30 @@ const (
...
-	return x.Success == struct{}{} && x.SystemError == 0 && x.UserDefinedError == 0
+func (x ExitCode) IsError() bool {
+	return !x.IsSuccess()
 }
 
-func (x *ExitCode) IsError() bool {
-	return !x.IsSuccess()
+func (x ExitCode) Equals(e ExitCode) bool {
+	return x == e
 }
```
Comment (anorth) [inline src/actors/runtime/exitcode/vm_exitcodes.go]: This whole function is probably unnecessary now

File: src/actors/runtime/exitcode/vm_exitcodes.go [hunk 3]
```diff
@@ -65,56 +64,30 @@ const (
...
-func (x *ExitCode) AllowsStateUpdate() bool {
+func (x ExitCode) AllowsStateUpdate() bool {
 	return x.IsSuccess()
 }
 
-func (x *ExitCode) Equals(ExitCode) bool {
-	panic("")
+func OK() ExitCode {
+	return ExitCode(0)
 }
```
Comment (anorth) [inline src/actors/runtime/exitcode/vm_exitcodes.go]: Replace this with a `const OK = ExitCode(0)`.

File: src/systems/filecoin_vm/interpreter/vm_interpreter.go [hunk 4]
```diff
@@ -136,8 +136,8 @@ func (vmi *VMInterpreter_I) ApplyMessage(inTree st.StateTree, chain chain.Chain,
 		retReceipt = vmri.MessageReceipt_Make(invocOutput, exitCode, vmiGasUsed)
 	}
 
-	_applyError := func(tree st.StateTree, errExitCode exitcode.SystemErrorCode, senderResolveSpec SenderResolveSpec) {
-		_applyReturn(tree, vmr.InvocOutput_Make(nil), exitcode.SystemError(errExitCode), senderResolveSpec)
+	_applyError := func(tree st.StateTree, errExitCode exitcode.ExitCode, senderResolveSpec SenderResolveSpec) {
```
Comment (anorth) [inline src/systems/filecoin_vm/interpreter/vm_interpreter.go]: As a follow-up, let's move this to a package with a less redundant name.

File: src/actors/runtime/exitcode/vm_exitcodes.go (hunk 2 above)
Comment (frrist) [reply src/actors/runtime/exitcode/vm_exitcodes.go]: True, but it's slightly more convenient than `if exitCode == Exitcode(foo) {`. I'll remove it if you'd like.

File: src/systems/filecoin_vm/interpreter/vm_interpreter.go (hunk 4 above)
Comment (frrist) [reply src/systems/filecoin_vm/interpreter/vm_interpreter.go]: added a TODO for this.

**Pull request merged**